### PR TITLE
Fix #361 Update redirect tests using unsupported locales

### LINKED ISSUE #361: Look at adding locale redirect-specific tests for /en/ and /es/ (to cover the edge case we missed in bug 1101261)
Apparently, even with the pretty good locale and redirect testing we do, we missed bug 1101261 -- let's take a look at how we can help ensure that for /en/ and /es/ we're setup to not miss that again.

@kyoshino added a fix and a test to a submodule in vendor, but let's see about adding coverage here, so we can hit dev, stage, and prod on Real Servers (tm) with real configs, etc.

### PR DESCRIPTION
**Pull request by kyoshino**
All tests passed locally.

### CODE REVIEW

**Comment by stephendonner** [comment]
@m8ttyB can you re-run these against dev/stage/prod, and if it passes, merge in?  Thanks.

File: tests/test_redirect_landing.py [hunk 1]
```diff
@@ -11,17 +11,25 @@
...
+        'af', 'an', 'ar', 'as', 'ast', 'az', 'be', 'bg', 'bn-IN', 'br', 'ca',
+        'cs', 'cy', 'da', 'de', 'dsb', 'el', 'en-GB', 'en-US', 'eo', 'es-AR',
+        'es-CL', 'es-ES', 'es-MX', 'et', 'eu', 'fa', 'ff', 'fi', 'fr', 'fy-NL',
+        'ga-IE', 'gd', 'gl', 'he', 'hi-IN', 'hr', 'hsb', 'hu', 'hy-AM', 'id',
+        'is', 'it', 'ja', 'ka', 'kk', 'km', 'ko', 'lij', 'lt', 'lv', 'mk', 'ml',
+        'mr', 'ms', 'my', 'nb-NO', 'nl', 'nn-NO', 'oc', 'pa-IN', 'pl', 'pt-BR',
+        'pt-PT', 'rm', 'ro', 'ru', 'sk', 'sl', 'son', 'sq', 'sr', 'sv-SE', 'ta',
+        'te', 'th', 'tr', 'uk', 'ur', 'uz', 'vi', 'xh', 'zh-CN', 'zh-TW',
     )
+    LOCALE_VARIANTS = {
```
Comment (m8ttyB) [inline tests/test_redirect_landing.py]: Small nit, can you add a code comment defining what `LOCALE_VARIANTS` is for and how it differs from `LOCALES`?

**Comment by m8ttyB** [comment]
I really like the refactor,  it's very clean.

There appears to one lingering failure when the tests are run, I ran them against production.

    14:52:57 ____________ TestRedirectLanding.test_redirect_firefox_using_locale ____________
    14:52:57 [gw2] linux2 -- Python 2.7.9 /var/lib/jenkins/jobs/mozilla.org.361-locale-redirects/workspace/.env/bin/python2.7
    14:52:57 tests/test_redirect_landing.py:91: in test_redirect_firefox_using_locale
    14:52:57 >           self._test_redirect(mozwebqa, '/firefox/', final, headers)
    14:52:57 tests/test_redirect_landing.py:44: in _test_redirect
    14:52:57 >       Assert.contains(final, response.url)
    14:52:57 .env/lib/python2.7/site-packages/unittestzero.py:191: in contains
    14:52:57 >           raise AssertionError('%s is not found in %s. %s' % (needle, haystack, msg))
    14:52:57 E           AssertionError: mozilla.com.tw/firefox/new/ is not found in https://www.mozilla.org/zh-TW/firefox/new/

**Comment by kyoshino** [comment]
Updated the tests with comments and the latest locale list.
`/zh-TW/firefox/new/` is no longer redirected to the Mozilla Taiwan site. ([Bug 1115626](https://bugzilla.mozilla.org/show_bug.cgi?id=1115626))

**Comment by m8ttyB** [comment]
r+ Many thank @kyoshino. Merging
